### ISSUE TITLE
create-cd-jenkins.sh: `type -P` does not work in all linux distros

### ISSUE THREAD

**Issue by hugowschneider** [opendevstack/ods-core#331]
https://github.com/opendevstack/ods-core/blob/0bbb48417085d15515c732fe45185581f74f9c1c/create-projects/create-cd-jenkins.sh#L9

does not work in all linux distros. Testing the script in some different environments like:

- Ubuntu Xenial in travis: https://travis-ci.com/opendevstack/ods-core/jobs/266490723#L630
```
+type -P tailor
+tailor_exe=-P: not found
tailor is /usr/local/bin/tailor
```
- MacOS
`
type: bad option: -P
`

**Comment by henrjk** [comment]
While I think this can be easily adjusted in this case. There could be other cases where it is more complicated to ensure wider compatibility.

Also on my mac `type -P` works:

```
~ $ type -P tailor
/usr/local/bin/tailor
~ $ type -P type
/usr/bin/type
```

For this script I assumed that compatibility with cd/jenkins-slave-base would be required.
Macos compatibility would be nice to have for sure as many opendevstack devs are mac users.

Nonetheless I am wondering what requirements there are regarding distro compatibility?

**Comment by hugowschneider** [comment]
It need to run in travis, that is the major pain point

**Comment by henrjk** [comment]
I did some research and found that `type -P` should be available on travis xenial version.
@hugowschneider and I verified this earlier today. 

Now it is less clear why this is not available when the command is executed using the following function [RunScriptFromBaseDir](https://github.com/BIX-Digital/ods-core/blob/eeff9495aab9afc23ef9b49044a12e1b3f8cd18e/tests/utils/command-utils.go#L15). 

I found a reference that go would intentionally not use the system shell (see https://stackoverflow.com/a/52281031). 

Inspired from this I also noted on a ubuntu 1804 virtual box machine:

```bash
~$ bash -ctype -P ls
bash: -y: invalid option
```
whereas

```bash
~$ bash -c type -P ls
~$ echo $?
0
```
